### PR TITLE
Prohibit calls to java.nio from UI thread, with reasonable exceptions

### PR DESCRIPTION
**Pull request by PawelLipski**
Spin-off to https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1152/

### CODE REVIEW

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/gitrepositoryselection/GitRepositoryComboBox.java [hunk 1]
```diff
@@ -56,8 +56,6 @@ public MutableCollectionComboBoxModel<GitRepository> getModel() {
 
   @UIEffect
   private void updateRepositories() {
-    // A bit of a shortcut: we're accessing filesystem even though we are on the UI thread here;
```
Comment (PawelLipski) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/gitrepositoryselection/GitRepositoryComboBox.java:59]: These comments are still valid; it's just that now all such cases are "managed centrally" via ArchUnit

File: src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java [hunk 2]
```diff
@@ -86,12 +86,66 @@ public void check(JavaMethod method, ConditionEvents events) {
         .check(importedClasses);
   }
 
+  private static final String[] whitelistedMethodFullNames_git4idea = {
```
Comment (PawelLipski) [inline src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java:89]: No changes to the list for git4idea, other than abc sorting ;p

File: src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java [hunk 3]
```diff
@@ -101,54 +155,19 @@ public void check(JavaMethod method, ConditionEvents events) {
...
-                    "git4idea.ui.ComboBoxWithAutoCompletion.setUI(javax.swing.plaf.ComboBoxUI)",
-                    "git4idea.ui.ComboBoxWithAutoCompletion.selectAll()",
-                    "git4idea.validators.GitBranchValidatorKt.checkRefName(java.lang.String)"
-                };
-                String calledMethodFullName = callTarget.getFullName();
-
-                if (!Arrays.asList(whitelistedMethodFullNames).contains(calledMethodFullName)) {
+                if (!Arrays.asList(whitelistedMethodFullNames_git4idea).contains(calledMethodFullName)) {
                   String message = "a non-${UIThreadUnsafeName} method ${method.getFullName()} " +
                       "calls method ${calledMethodFullName} from git4idea";
```
Comment (mkondratek) [inline src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java:162]: so soon the plugin name will be vcs-git but the package will remain git4diea, right?

**Review by mkondratek — APPROVED** [review]

File: src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java (hunk 3 above)
Comment (PawelLipski) [reply src/test/java/com/virtuslab/archunit/UiThreadUnsafeMethodInvocationsTestSuite.java:162]: :yes:, exactly

**Pull request merged**